apps/main/lib/main/repositories/articles_repo.rb

```diff
@@ -0,0 +1,14 @@
+require "blog/repository"
+require "main/entities/article"
+
+module Main
+  module Repositories
+    class ArticlesRepo < Blog::Repository[:articles]
+      struct_namespace Main::Entities
+
+      def home_page_listing
```

Comment:
You're right, we could probably just call this `#listing` and accept limit as an argument.